lib/dialects/mssql/query.js

```diff
@@ -211,6 +211,10 @@ class Query extends AbstractQuery {
           primaryKey: _result.Constraint === 'PRIMARY KEY',
           autoIncrement: _result.IsIdentity === 1
         };
+
+        if (result.Length !== null) {
+          result[_result.Name].type += `(${_result.Length})`;
```

Comment:
If check should be testing for string types, if its a string type, find length and append it properly